### DISCUSSION TITLE
Support Enlightn's --allow-list option

### DISCUSSION THREAD

**Discussion by phil-s** [phpro/grumphp#1037] [Ideas]
| Q               | A
| --------------- | ---
| Version         | GrumPHP 1.3.2 (but looks to be the same in master)
| Bug?            | no
| New feature?    | yes
| Question?       | yes
| Documentation?  | yes
| Related tickets | n/a

https://github.com/phpro/grumphp/blob/master/doc/tasks/securitychecker/enlightn.md indicates that the only options for configuring `enlightn` are `lockfile` and `run_always`; and my reading of `SecurityCheckerEnlightn::run()` and friends (including `ProcessArgumentsCollection.php`) seems to agree.

I'd like to be able to configure Enlightn's `--allow-list` option first and foremost.  Can we get support for that?

Refer to: https://github.com/enlightn/security-checker#allow-vulnerabilities

(More generally, it seems as if it would be tremendously useful to have a task-agnostic way of configuring additional command-line options to pass to processes, so that such things can be done with ease even when grumphp doesn't provide specific support for an option.)

**Comment by veewee** [comment]
I'dd surely accept a PR for that.

On the second part:
It has been asked before to provide additional command-line options.
However, due to the structure of the tasks - it is not something that can easily be applied - since it needs to be configured and implemented manually for every external task out there ATM. If you find an elegant solution for the problem, we could go for that. But maybe best to start a new discussion before implementing.

**Comment by peterjaap** [comment]
+1 for this feature. Came here to look for it, but found this discussion instead.

**Reply by peterjaap** [reply]
I opened a PR for it https://github.com/phpro/grumphp/pull/1052
